### ISSUE TITLE
Add starfetch to homebrew

### ISSUE THREAD

**Issue by dannydorazio** [Haruno19/starfetch#1]
Hi!  I love starfetch and can't wait to see it grow.  I was wondering if you had looked at publishing to homebrew?  This would make it much easier to both install and update.  Thanks for your consideration!

https://docs.brew.sh/Adding-Software-to-Homebrew

**Comment by Haruno19** [comment]
Thank you! I was already considering publishing starfetch to Homebrew in the future. Maybe I'll make the code more efficient and add some more constellation, and then I'll add it to Homebrew!

**Comment by mBaratta96** [comment]
Thank you for this wonderful tool! I would also like to suggest adding a repository for Linux to easily receive updates.

**Comment by Haruno19** [comment]
Now that the code has been revised, I'm currently looking into adding starfetch to Hombrew and other repositories. I'm still very new to this so I have to figure out how to do so.

In the meantime, a Pull Request to add starfetch to Nixpkgs has already been opened [here](https://github.com/NixOS/nixpkgs/pull/167541) thanks to user [papojari](https://github.com/papojari).

**Comment by tildethinexxbisense** [comment]
how's the progress going? I have plenty of people who would use it if its on the APT because they're too afraid of building from source lol!

**Comment by Haruno19** [comment]
Yeah I’m so sorry it’s taking long! I’m now writing the formulae file to add it to homebrew but there’s still something wrong… I’m totally clueless with this kind of things, I’m sorry :(

**Comment by papojari** [comment]
@06kellyjac would like git versioning for packaging

**Comment by 06kellyjac** [comment]
Not required but advised. This is what I put in the review

> Since the maintainer is interested in this being packaged id advise they tag a version 0.0.1
People are more likely to package a release than a commit

**Comment by Haruno19** [comment]
So, I have to create a release tagging it as version 0.0.1?
(I'm very sorry for my ignorance about this)

**Comment by papojari** [comment]
You don't have to but it would be nice. It also doesn't have to  be 0.0.1.

**Comment by 06kellyjac** [comment]
@papojari is exactly right

---

Just some extra detail. A tag is a git feature a bit like a branch fixed in time. A github release can then be created from a tag if you want but just a tag is enough for us

Also I suggested 0.0.1 as a first version due to something called SemVer all the 0.X.X versions can have breaking changes, once you have a numbered release like 1.0.0 or 2.4.3 people might expect there to be "no breaking changes"

https://semver.org/

Heres how to make a tag

https://stackoverflow.com/questions/18216991/create-a-tag-in-a-github-repository

**Comment by Haruno19** [comment]
Ok thank you! 
I've just published a release with tag 0.0.1.

**Comment by papojari** [comment]
I tried to package 0.0.1 with Nix but it doesn't work properly #10

**Comment by auroraanna** [comment]
@dannydorazio you can close this issue. starfetch has been added for quite some time now.

**Comment by dannydorazio** [comment]
The original reason I opened this issue was for it to be added to Homebrew, which I don't believe has been added yet-- unless I am missing something.

**Comment by auroraanna** [comment]
Oops, I assumed this is the nixpkgs repo…
Well now you know that it has been added to nixpkgs I guess.

**Comment by auroraanna** [comment]
I think nixpkgs is way better than Homebrew tho >:P And you can have nixpkgs on MacOS via Home manager.

**Comment by DtxdF** [comment]
> Now that the code has been revised, I'm currently looking into adding starfetch to Hombrew and other repositories. I'm still very new to this so I have to figure out how to do so.
> 
> In the meantime, a Pull Request to add starfetch to Nixpkgs has already been opened [here](https://github.com/NixOS/nixpkgs/pull/167541) thanks to user [papojari](https://github.com/papojari).

I think this is not necessarily the responsibility of the starfetch developers. Of course, if they can port starfetch to a variety of platforms, that's good, but the person who likes and uses starfetch, and knows how to port to their favorite platform, is the person who has the responsibility for such a task. That person is not necessarily a starfetch developer.

> @auroraanna The original reason I opened this issue was for it to be added to Homebrew, which I don't believe has been added yet-- unless I am missing something.

I respect that the main topic of this thread is for homebrew, but I think this is useful to comment on the progress on porting starfetch on a variety of platforms.

If it is a help for this project, I have [ported](https://www.freshports.org/astro/starfetch/) starfetch to FreeBSD.

**Comment by auroraanna** [comment]
> but the person who likes and uses starfetch, and knows how to port to their favorite platform, is the person who has the responsibility for such a task. That person is not necessarily a starfetch developer.

fair

**Comment by jacobmealey** [comment]
Hi folks, I just wanted to let you all know I've started the process for getting starfetch into the debian repos: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1123851

**Comment by jacobmealey** [comment]
Hello,
As a follow up to my previous comment, starfetch has been accepted into Debian unstable today and should be available in testing/forky soon. 

https://tracker.debian.org/pkg/starfetch

**Comment by Haruno19** [comment]
@jacobmealey thank you so much!! 😄
